MBL-11865: Initial inclusion of android espresso and docs automation dirs

joehoag opened the pull request:
I tried to make it as minimalistic as possible.  I would assume that the main thing that you would want to look at would be build.gradle and other configuration-related files.  All actual code files have been copied verbatim from android-uno.

joehoag committed:
MBL-11865: Initial inclusion of android espresso and docs automation dirs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/instructure/mobile-shared?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/instructure/mobile-shared?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/instructure/mobile-shared?pullRequest=1) it.</sub>

joehoag commented on automation/android/espresso/build.gradle:
```diff
@@ -0,0 +1,197 @@
+apply plugin: 'kotlin-android'
+
+// Borrowed from http://stackoverflow.com/questions/17097263/automatically-versioning-android-project-from-git-describe-with-android-studio-g
+def getVersionCode = {
```
My attempt at automatically producing a version number.

bootstraponline replied:
we may want to consistently use the commit hash, I'm not sure there's an easy way to resolve build numbers to git commits.

joehoag replied:
This has worked for me in the past.  And commit hashes wouldn't be ordered, right?

joehoag replied:
I guess that my thinking is that we'll tag each bitrise build with the bitrise build number, and that will be used to identify whatever we push to jitpack.  I suppose that we could manually maintain the build number, like we do with our canvas apps, but I'd like to see if this works.

bootstraponline replied:
The hash becomes the build number. Checkout how Java projects publish snapshots on jitpack. It's usually something like Flank-master-f40414091f-1.jar

bootstraponline replied:
> And commit hashes wouldn't be ordered, right?

they're ordered in the git history, that's about it. GitHub will show you how many commits behind a commit is.

joehoag replied:
Apologies for my slowness, but how is that better than Flank-master-1, Flank-master-2, Flank-master-3, ...?

bootstraponline replied:
let's chat more on Slack, I think there are a few reasons why that's not common practice on jitpack.

joehoag commented on automation/android/espresso/gradle.properties:
```diff
@@ -0,0 +1,4 @@
+org.gradle.jvmargs=-Xmx2560M
+org.gradle.caching=true
+android.useAndroidX=true
+android.enableJetifier=true
```
Lines 3 and 4 did the jetifier thing.  I'm hoping that this works in bitrise as well as Android Studio.

bootstraponline replied:
yep it should

joehoag commented:
It looks like bitrise needs the gradle wrapper to be present.  Maybe I should un-ignore gradle/ and gradlew*?

bootstraponline commented:
Yeah, the Gradle wrapper is what bitrise calls

bootstraponline commented on automation/android/espresso/build.gradle:
```diff
@@ -0,0 +1,197 @@
+
+buildscript {
+    // Define this once here because it will need to be reused later
+    ext.kotlinVersion = "1.2.70"
```
this version of kotlin is now quite old, we'll want to update to the latest stable

joehoag replied:
Does that mean that we would want to up-version Kotlin in android-uno as well, to keep the versions in sync?

bootstraponline replied:
Ideally yes, we'd want android-uno on the latest Kotlin version. Upgrading android-uno is typically more challenging because there's a lot of code.

We should be able to use the newer version on this lib and still use it from an older Kotlin in android-uno.

bootstraponline commented on automation/android/espresso/build.gradle:
```diff
@@ -0,0 +1,197 @@
+}
+
+def _compileSdkVersion = getProperty('compileSdkVersion', 26)
+def _buildToolsVersion = getProperty('buildToolsVersion', '28.0.3')
```
these getProperty calls can all be removed, they're legacy code.

bootstraponline commented on automation/android/espresso/build.gradle:
```diff
@@ -0,0 +1,197 @@
+    //
+    // https://maven.google.com/com/android/support/test/espresso/espresso-core/3.0.0/espresso-core-3.0.0.pom
+
+    def runnerVersion = "1.1.0-beta01"
```
these versions look out of date, there should be new versions for the espresso dependencies

joehoag replied:
This is what we had defined in our android-uno Versions object.  I'll look around to see if there are better/newer versions.  (Note that the Android Studio is not complaining that there versions are out of date, as it seems to normally do.)

bootstraponline replied:
> (Note that the Android Studio is not complaining that there versions are out of date, as it seems to normally do.)

right, it won't do that for any dynamically defined versions.

bootstraponline replied:
find the latest stable versions here:
https://developer.android.com/training/testing/release-notes

joehoag committed:
MBL-11865: Addressed PR feedback.

- Un-ignored gradle wrapper files
- Upgraded versions of some espresso/kotlin libraries
- Removed some obsolete code from build.gradle.
- Removed build versioning logic from build.gradle.
